File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int lo = 0;
        int hi = nums.length-1;
        while(lo <= hi){
            int mid =  lo + (hi - lo)/2;
            if(nums[mid] == target){
                return mid;
            }
            else if(nums[mid] > target){
                hi = mid - 1;
            }
            else{
                lo = mid+1;
            }
        }
        return -1;
    }
}

File: 792-binary-search/binary-search.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        var lo = 0
        var hi = nums.size-1
        while(lo<=hi){
            val mid = lo + (hi-lo)/2
            when{
                nums[mid]==target -> return mid
                nums[mid] < target -> lo = mid+1
                nums[mid] > target -> hi = mid-1
            }
        }
        return -1
    }
}